### ISSUE TITLE
[physac] physics_demo and physics_shatter crash when hitting R to reset example

### ISSUE THREAD

**Issue by pamarcos** [raysan5/raylib#592]
Both `physics_demo` and `physics_shatter` crash when hitting `R` to reset example. Sometimes `physics_shatter` gets creative instead of crashing directly. It ends up crashing, eventually

![image](https://user-images.githubusercontent.com/3248441/43196009-ef58af50-9006-11e8-8c6b-421919b58d22.png)

**Comment by raysan5** [comment]
Maybe @victorfisac could take a look to this issue...

**Comment by drezgames** [comment]
This happens because the code as it is, is not thread safe. You hit reset to clear the data while the background thread is updating data, you will get these random crashes. I use Object Pascal so for me, I had to disable threading on the C side and manage threading on the Pascal side. To make it work, I just put some mutex calls around the routines that change data. It seems to work and not crash anymore.

**Comment by raysan5** [comment]
Maybe multithreading logic should be moved out of `physac` and let the user implement it on their end...

**Comment by drezgames** [comment]
yea, maybe best thing because just 1 min ago, I got it working smooth, no crashes and less cpu usage. Up until just now I got the crashing fixed, but it was suffering from a little stuttering and it took a simple change to fix that. So now it seems to working good. In my threading loop, just need to do Sleep(1) to allow other threads to run better and that was it. Now no stuttering as the mutex is locked while destroying bodies.
